### ISSUE TITLE
Parsing fails on apple-silicon/arm64, wind and windows

### ISSUE THREAD

**Issue by MariusMaibornWolff** [MaibornWolff/codecharta#2953]
# Bug

gitlogparser, rawtextparser and sourcecodeparser fail on Apple Silicon.

The command gets stuck on apple silicon with no further output.
No output file is written after executing the command.
e.g.:

ccsh gitlogparser "git_output" "--output-file=gitlogparserresult.json" "--file-name-list=git_filename_list_output" "--not-compressed" "--silent=false" "--add-author=true"

output:

 Assumed encoding UTF-8
  57% [=========================================================>                                           ]  9868725/17018092 Byte, ETA: 00:00:00
Unhandled Edge Case in MergeCommit: deleted: false, mutated: true, modification type: RENAME, initalAdd: false, file: apps/community-service/.idea/runConfigurations.xml
  58% [==========================================================>                                          ]  9881689/17018092 Byte, ETA: 00:00:00
Unhandled Edge Case in MergeCommit: deleted: false, mutated: true, modification type: RENAME, initalAdd: false, file: apps/community-service/.idea/runConfigurations.xml
 100% [====================================================================================================>] 17018092/17018092 Byte, ETA: 00:00:00
Created Project with 3325 leaves.

## Expected Behavior

Process finishes and output json file has been created.

## Actual Behavior

Process gets stuck. No further output/error is given.

## Steps to Reproduce the Problem
1. User a mac with apple-silicon / arm64
2. Run any parser on any project
3. Observe that the process is stuck and no further output is given.

## Specifications
- is released in [online-demo](https://maibornwolff.github.io/codecharta/visualization/app/index.html?file=codecharta.cc.json&file=codecharta_analysis.cc.json): yes | no
- CodeCharta Version: latest
- OS: Mac OS 
- Browser: Chrome

**Comment by MariusMaibornWolff** [comment]
For solution: Eventually check if parsers run in a docker container successfully.

**Comment by ce-bo** [comment]
@Hall-Ma found out, that this applies also for Windows (tested on Windows with ccsh version 1.103.3).

**Comment by ce-bo** [comment]
Let's extend `golden_test.sh` to have an automated regression test in case this bug reoccurs sometime.

**Comment by Hall-Ma** [comment]
When you are trying to use Sonar Importer on Windows (Git Bash) you get no output file because the script doesn't end:
![image](https://user-images.githubusercontent.com/72517530/184597952-9dacf16b-a719-487f-83a5-d4d6aace13b1.png)
I tried to debug a bit and found out that the method `wrapInput` of the new `CompressedStreamHandler` stuck when it tries to read the content `val firstValue = content.read()`. @phanlezz Do you have any idea why?
On Mac, the Sonar Importer works.

**Comment by phanlezz** [comment]
> When you are trying to use Sonar Importer on Windows (Git Bash) you get no output file because the script doesn't end: ![image](https://user-images.githubusercontent.com/72517530/184597952-9dacf16b-a719-487f-83a5-d4d6aace13b1.png) I tried to debug a bit and found out that the method `wrapInput` of the new `CompressedStreamHandler` stuck when it tries to read the content `val firstValue = content.read()`. @phanlezz Do you have any idea why? On Mac, the Sonar Importer works.

We had this issue before and we added a workaround. I'm not quite sure why it stopped working again. I might be able to look later today or tomorrow into it. 😞 
This issue is/was that the programm doesn't know if there is something incoming on the` System.in` . I added the `.available `check, but thats an old (and like we now see) unrealiable method of checking if there is an input steam as it seems. 
The root of the problem is method later in the scipt, that checks `System.In` always one second after calling it. There is an open issue explaining this behavior: #2858 and the workaround #2854 (where we found this problem).
We might be able to remove the compression handler from the deserializer that handles generic input streams, but for that we would need to rewrite almost every test which rely often on the input stream method to parse sample project files (since these could not be transferred as a FileInputStream) 

The git bash seems to act differently than cmd or other CLIs.

Small workaround for the moment, if it is an urgent problem: CTRL + D (normally sends EOF to console)
